app/src/main/java/com/cortado/minionify/ModeChoiceActivity.java

```diff
@@ -0,0 +1,75 @@
+package com.cortado.minionify;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.cortado.minionify.accumulators.IAccumulator;
+import com.cortado.minionify.accumulators.PreferencesFiles;
+import com.cortado.minionify.accumulators.Preferences;
+import com.cortado.minionify.mode.AppMode;
+import com.cortado.minionify.accumulators.PreferenceAccumulator;
+
+public class ModeChoiceActivity extends AppCompatActivity {
+
+    private IAccumulator mAccumulator;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mAccumulator = new PreferenceAccumulator(getSharedPreferences(PreferencesFiles.GENERAL_PREFERENCES, Context.MODE_PRIVATE));
+        redirectActivityIfModeExists();
+
+        Window window = getWindow();
```

Comment:
you can combine these 3 lines to "setupGui" (choose a  better name)

Reply:
Done